### PR TITLE
Update to SDK 2.24.0

### PR DESCRIPTION
**Pull request by lawmicha**
*Issue #, if available:*

*Description of changes:*
Update to SDK 2.24.0

- [x] Integ test "AWSAppSync-IntegTests" target has passed
- [x] unit test has passsed

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: Package.swift [hunk 1]
```diff
@@ -14,7 +14,7 @@ let package = Package(
         .package(
             name: "AWSiOSSDKV2",
             url: "https://github.com/aws-amplify/aws-sdk-ios-spm.git",
-            .upToNextMinor(from: "2.23.3")
+            .upToNextMinor(from: "2.24.0")
```
Comment (lawmicha) [inline Package.swift:17]: first SPM related update

File: Package.resolved [hunk 2]
```diff
@@ -15,8 +15,8 @@
         "repositoryURL": "https://github.com/aws-amplify/aws-sdk-ios-spm.git",
         "state": {
           "branch": null,
-          "revision": "9057509ddfc1f3ef1c36264e9b8e8173eda634c9",
-          "version": "2.23.3"
+          "revision": "7be6c13ea6fdcfdf54d50c08a10928e2e0bc3fb0",
+          "version": "2.24.0"
```
Comment (lawmicha) [inline Package.resolved:19]: generated from `swift package resolve`

File: Package.swift (hunk 1 above)
Comment (lawmicha) [reply Package.swift:17]: like the cart file, we should be able to add this under the script

Comment (royjit) [reply Package.swift:17]: This was manually added right?

**Review by royjit — APPROVED** [review]
LGTM

File: Package.swift (hunk 1 above)
Comment (lawmicha) [reply Package.swift:17]: yes, will open separate PR for updating script

Comment (lawmicha) [reply Package.swift:17]: oh wait, it's already in the script, my bad

**Pull request merged**